python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java

```diff
@@ -287,11 +287,12 @@ private FunctionType buildFunctionType(FunctionDef functionDef) {
       functionTypeBuilder.withTypeOrigin(TypeOrigin.LOCAL);
     }
     FunctionType functionType = functionTypeBuilder.build();
-    if (owner != null) {
+
+    if (owner instanceof ClassType classTypeOwner) {
```

Comment:
as discussed: this condition excludes adding members to the owner if it is `ModuleType` so probably this needs to be changed